source/byte_buf.c

```diff
@@ -665,29 +665,39 @@ bool aws_byte_cursor_satisfies_pred(const struct aws_byte_cursor *source, aws_by
 }
 
 int aws_byte_cursor_compare_lexical(const struct aws_byte_cursor *lhs, const struct aws_byte_cursor *rhs) {
-
+    AWS_PRECONDITION(aws_byte_cursor_is_valid(lhs));
+    AWS_PRECONDITION(aws_byte_cursor_is_valid(rhs));
+    AWS_PRECONDITION(AWS_MEM_IS_READABLE(lhs->ptr, lhs->len));
```

Comment:
As this is included in `aws_byte_cursor_is_valid` when `len > 0`, I think it is important to at least include a comment here that explains the intent of this precondition.